### PR TITLE
Allow dbal version 3

### PR DESCRIPTION
**Pull request by rabauss**
... to fix compatiblity to contao 4.13 😉

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -33,7 +33,7 @@
     "php": "^7.1 || ^8.0",
     "ext-pdo": "*",
     "contao/core-bundle": "^4.4",
-    "doctrine/dbal": "^2.5",
+    "doctrine/dbal": "^2.5 || ^3.0",
```
Comment (dmolineus) [inline composer.json]: Please increase requirements of version 2 to 2.11 as there the forward compatibility layer exists.

```suggestion
    "doctrine/dbal": "^2.11 || ^3.0",
```

**Review by dmolineus — CHANGES_REQUESTED** [review]
Allowing doctrine/dbal 3 alone is not enough. We need to adjust the usage of the doctrine dbal api here: https://github.com/hofff/contao-social-tags/blob/50231ef3d8cd1de1f68bf5d11c4405916c20f928/src/Data/SocialTagsFactory.php#L114

Could you please rebase to latest master (which will fix the current broken CI) also.

**Comment by rabauss** [comment]
Okay I adjusted the deprecated dbal usages.
By the way there are also some contao deprecations but therefor the dependencies need to be increased to contao `^4.9`

File: src/Data/SocialTagsFactory.php [hunk 2]
```diff
@@ -109,9 +108,9 @@ private function loadReferencePage(array $pageTrail, array $modes): ?PageModel
             ->setParameter('trailSet', $trailSet)
             ->setParameter('modes', $modes, Connection::PARAM_STR_ARRAY)
             ->setMaxResults(1)
-            ->execute();
+            ->executeQuery();
```
Comment (dmolineus) [inline src/Data/SocialTagsFactory.php]: Unfortunately the `executeQuery` is not part of the forward compatibility layer in doctrine/dbal v2. We have to stick to the `execute` method here, see https://github.com/doctrine/dbal/blob/2.13.x/lib/Doctrine/DBAL/Query/QueryBuilder.php

**Review by dmolineus — CHANGES_REQUESTED** [review]

**Comment by dmolineus** [comment]
Thanks for the contributions.

> Okay I adjusted the deprecated dbal usages.

I added a comment above, the unfortunately have to keep the deprecated `execute` method of the query builder. It got deprcated in `3.0` only and has to be used for BC.

>  By the way there are also some contao deprecations but therefor the dependencies need to be increased to contao ^4.9

We're fine bumping the requirement to Contao ^4.9. Feel free to open another PR to address these deprecations.

File: src/Data/SocialTagsFactory.php (hunk 2 above)
Comment (rabauss) [inline src/Data/SocialTagsFactory.php]:
```suggestion
            ->execute();
```

**Comment by dmolineus** [comment]
Thanks @rabauss

**Comment by dmolineus** [comment]
Released as 1.1.3

**Pull request merged**